### DISCUSSION TITLE
Proposal for Enhancing `themeToVars` Function to Include `other` Key Variables

### DISCUSSION THREAD

**Discussion by murphyjoseph** [mantinedev/mantine#6512] [Feature requests]
Apologies if this is somewhere else, I did a search for `themeToVars` and nothing else came up.

**Proposal**
I would like to propose an enhancement to the `themeToVars` function in the Mantine `@mantine/vanilla-extract library`. Currently, this function does not include the CSS variables defined under the `other` key in the custom theme. Including these variables would provide a more comprehensive and flexible theming experience.

**Motivation**
The `other` key in the custom theme is a useful place to store various theme-related values that do not fit into the predefined keys. By including the CSS variables from the other key in the output of `themeToVars`, developers can leverage the full potential of Mantine's theming system and maintain consistency across their applications.

**Example**
Here's an example to demonstrate the current behavior and the desired behavior after the proposed enhancement.

Current Behavior:
```js
import { themeToVars, createTheme } from '@mantine/core';

const customTheme = createTheme({
  colors: {
    primary: '#3498db',
    secondary: '#2ecc71',
  },
  other: {
    customVariable: '10px',
  },
});

const cssVariables = themeToVars(customTheme);
console.log(cssVariables);
// Output: --mantine-color-primary: #3498db; --mantine-color-secondary: #2ecc71;
// Desired output: --mantine-color-primary: #3498db; --mantine-color-secondary: #2ecc71; --mantine-other-customVariable: 10px;
```

**Benefits**
- Flexibility: Developers can define and use custom CSS variables in their themes.
- Consistency: Ensures that all theme-related variables are included in the generated CSS variables, leading to a more consistent theming experience.
- Ease of Use: Simplifies the process of accessing and using custom theme variables in CSS.

**Conclusion**
Incorporating the `other` key into the `themeToVars` function will enhance the flexibility and usability of the Mantine theming system. I believe this enhancement will be beneficial for many developers using Mantine, especially those like me who customize and build off of Mantine's out of the box theme schema.

Thank you for considering this proposal. I look forward to your feedback.

**Comment by rtivital** [comment]
`theme.other` is meant to store everything you might need in your theme. It is not strictly designed for CSS variables – you can put functions, numbers, objects and any other values there. If you need to define CSS variables in your application, it is better to do that in CSS file – it is more performant and you can choose any names for your variables.
